storage/store.go

```diff
@@ -1269,10 +1269,61 @@ func (s *Store) ReplicaCount() int {
 // ExecuteCmd fetches a range based on the header's replica, assembles
 // method, args & reply into a Raft Cmd struct and executes the
 // command using the fetched range.
-func (s *Store) ExecuteCmd(ctx context.Context, args proto.Request) (proto.Response, error) {
+func (s *Store) ExecuteCmd(ctx context.Context, args proto.Request) (reply proto.Response, _ error) {
 	ctx = s.Context(ctx)
 	trace := tracer.FromCtx(ctx)
-	// If the request has a zero timestamp, initialize to this node's clock.
+	if trace == nil {
+		trace = s.ctx.Tracer.NewTrace(args.Header())
+		ctx = tracer.ToCtx(ctx, trace)
+		defer trace.Finalize()
+	}
+	if batch, isBatch := args.(*proto.BatchRequest); isBatch {
+		// TODO(tschottdorf): provisional batch handling.
+		bReply := &proto.BatchResponse{}
+		var isTxn bool
+		if txn := batch.Txn; txn != nil {
```

Comment:
By the way, all of this Txn updating should really not be useful. We need to enforce that batches call EndTransaction only once and only at the end of the batch, which is the only time you get a Txn object back in the reply header which requires updating your current Txn.